pinot-common/src/main/java/org/apache/pinot/common/protocols/SegmentCompletionProtocol.java

```diff
@@ -149,6 +149,7 @@ public enum ControllerResponseStatus {
   public static final String REASON_END_OF_PARTITION_GROUP = "endOfPartitionGroup";
   // Stop reason sent by server as force commit message received
   public static final String REASON_FORCE_COMMIT_MESSAGE_RECEIVED = "forceCommitMessageReceived";
+  public static final String REASON_NUM_OF_COL_VALUES_ABOVE_THRESHOLD = "numColValuesAboveThreshold";
```

Comment:
This should be always on. We may introduce a config to turn it off if we are not confident about this new logic, but if it is not very complicated we can remove this config

Reply:
Considering new impln now where we are keeping conservative size estimation check as well, I guess it makes sense to enable this change only behind a flag?